sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -1151,10 +1151,21 @@ def extend(
         # When a corrupt value made it into the cache, we have to fail.
         raise Exception("Unexpected cached value: %s" % cached_value)
     # Write to state handler
+    futures = []
     out = coder_impl.create_OutputStream()
     for element in elements:
       coder.encode_to_stream(element, out, True)
-    return self._underlying.append_raw(state_key, out.get())
+      if out.size() > data_plane._DEFAULT_SIZE_FLUSH_THRESHOLD:
```

Comment:
This may overshoot the limit since it is encoding first.  Is there a cheap way to get the size of the encoded element to possibly flush before encoding to the stream? For dataflow (and perhaps other runners) there is a strict batch limit.  As written the largest safest element is that (strict limit - the flush threshold) instead of just the strict limit.

Reply:
There is not (in general, depends on the coder) a cheap way to get the size of the encoded element before actually encoding it. Given that the flush limit << the strict limit, I don't think this delta is worth optimizing for. One is already in risky territory getting close to the strict limit anyway.